src/Controllers/FoodVoucherController.php

```diff
@@ -0,0 +1,238 @@
+        }
+        if (!$this->userIsForced() && UserVouchers::eligibleVoucherCount($this->auth->user()) == 0) {
+            throw new HttpForbidden();
+        }
+    }
+
+    private function checkConfig(): void
+    {
+        $conf = $this->config->get('food_voucher_api');
+        if (
+            !(
+            $conf
+            && $conf['info_url']
+            && $conf['auth_token']
+            && $conf['post_url']
+            )
+        ) {
+            throw new HttpNotFound();
+        }
+    }
+
+    private function getAuthToken(): string
+    {
+        return $this->config->get('food_voucher_api')['auth_token'];
+    }
+
+    private function userIsForced(): bool
+    {
+        $user = $this->auth->user();
+        return $user->state->force_active && $this->config->get('enable_force_active')
+            || $user->state->force_food && $this->config->get('enable_force_food');
+    }
+
+    /**
+     * @throws ErrorException
+     */
+    private function getInfo(bool $crew, ?array $userMealVouchers = null): array
+    {
+        $this->checkConfig();
+
+        $data = cache('foodVoucherInfo', function () {
+            $infoUrl = (string) $this->config->get('food_voucher_api')['info_url'];
+            try {
+                $response = $this->guzzle->get(
+                    $infoUrl,
+                    ['headers' =>
+                        [
+                            'Authorization' => 'Bearer ' . $this->getAuthToken(),
+                            'Content-Type' => 'application/json',
+                        ],
+                    ]
+                );
+            } catch (ConnectException | GuzzleException $e) {
+                $this->log->error('Exception during food voucher api request', ['exception' => $e]);
+                throw new ErrorException('user.food.request-error');
+            }
+            if ($response->getStatusCode() !== 200) {
+                throw new HttpNotFound('user.food.request-error');
+            }
+            return json_decode($response->getBody()->getContents(), true);
+        }, 60);
+
+        $now = Carbon::now();
+        $locale = $this->translator->getLocale();
+        $futureMeals = [];
+        $gotMeals = [];
+
+        foreach ($data as $id => $meal) {
+            $endTime = Carbon::parse($meal['datetime']['date'] . ' ' . $meal['datetime']['end']);
+            if ($now < $endTime) {
+                $sold_out = $crew
+                    ? $meal['availability']['crew']
+                    : $meal['availability']['regular'];
+                $sold_out = $sold_out === false || $sold_out === 'false';
+
+                $futureMeals[$id] = [
+                    'id' => $id,
+                    'name' => $locale === 'de_DE' ? $meal['name']['de'] : $meal['name']['en'],
+                    'endtime' => $endTime,
+                    'sold_out' => $sold_out,
+                    'hidden' => $userMealVouchers && in_array($id, $userMealVouchers),
```

Comment:
Imho in that case it should be directly skipped with `continue;`

Reply:
my solution is a bit easier even though it is uglier